storage/remote/queue_manager.go

```diff
@@ -158,6 +168,13 @@ func NewStorageQueueManager(tsdb StorageClient, cfg *StorageQueueManagerConfig)
 // sample on the floor if the queue is full.
 // Always returns nil.
 func (t *StorageQueueManager) Append(s *model.Sample) error {
+	s.Metric = model.Metric(
```

Comment:
Maybe move the external labels down here.

There's correctness stuff around relabelling that's harder to verify if the code is spread around.